app/scripts/lib/nonce-tracker.js

```diff
@@ -12,15 +12,14 @@ class NonceTracker {
   // releaseLock must be called
   // releaseLock must be called after adding signed tx to pending transactions (or discarding)
   async getNonceLock (address) {
+    const pendingTransactions = this.getPendingTransactions(address)
     // await lock free
-    await this.lockMap[address]
+    if (pendingTransactions.length) await this.lockMap[address]
+    else if (this.lockMap[address]) await this.lockMap[address]()
     // take lock
     const releaseLock = this._takeLock(address)
     // calculate next nonce
-    const currentBlock = await this._getCurrentBlock()
-    const blockNumber = currentBlock.number
-    const pendingTransactions = this.getPendingTransactions(address)
-    const baseCount = await this._getTxCount(address, blockNumber)
+    const baseCount = await this._getTxCount(address)
```

Comment:
the order of requests here is really important:
- get current block
- get pending transactions
- get base tx count @ block
- calculate new nonce
we should add comments explaining why this is:
> we need to make sure our `base count` and `pending count` are from the same block